src/go/rpk/pkg/cli/cmd/cluster/config/lint.go

```diff
@@ -0,0 +1,115 @@
+in earlier versions of redpanda, but are now managed via Redpanda's
+central configuration store (and via 'rpk cluster config edit').
+
+This command is not for general changes to cluster configuration: use this only
+when redpanda will not start due to a configuration issue.
+
+This command erases a named property from the local node's internal 
+cache, so that on next startup redpanda will treat the setting as if it 
+was set to the default.
+`,
+		Run: func(cmd *cobra.Command, propertyNames []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			schema, err := client.ClusterConfigSchema()
+			out.MaybeDie(err, "unable to query config schema: %v", err)
+
+			configFile, err := p.LocateConfig(fs)
+			// The LocateConfig error type is a full explanation, pass it through
+			// without qualification.
+			out.MaybeDie(err, "%v", err)
+
+			configIn, err := afero.ReadFile(fs, configFile)
+			out.MaybeDie(err, "unable to read config file %q: %v", configFile, err)
+
+			// Decode into intermediate yaml.Node representation to preserve comments
+			// when writing the linted file back out.
+			inputDoc := make(map[string]yaml.Node)
+			err = yaml.Unmarshal(configIn, inputDoc)
+			out.MaybeDie(err, "unable to parse config file: %v", err)
+
+			cleanedDoc := make(map[string]yaml.Node)
+			var cleanedProperties []string
+
+			for key, node := range inputDoc {
+				if key == "redpanda" {
+					var cleanedContent []*yaml.Node
+					for i := 0; i < len(node.Content); i += 2 {
+						keyNode := node.Content[i]
+						valNode := node.Content[i+1]
+						key := keyNode.Value
+						_, is_cluster_config := schema[key]
+						if is_cluster_config {
+							cleanedProperties = append(cleanedProperties, key)
+						} else {
+							cleanedContent = append(cleanedContent, keyNode)
+							cleanedContent = append(cleanedContent, valNode)
+						}
+					}
+					node.Content = cleanedContent
+				}
+
+				cleanedDoc[key] = node
+			}
+
+			if len(cleanedProperties) > 0 {
+				log.Info("Cleaned up cluster configuration properties:")
+				for _, key := range cleanedProperties {
+					log.Infof(" - %s", key)
+				}
+				log.Info("") // Blank line
+			}
+
+			configOut, err := yaml.Marshal(cleanedDoc)
+			out.MaybeDie(err, "error serializing config: %v", err)
+
+			backupFileName := filepath.Base(configFile) + ".bak"
+			backupFile := filepath.Join(filepath.Dir(configFile), backupFileName)
+			log.Infof("Backed up configuration file to %q", backupFileName)
+			err = afero.WriteFile(fs, backupFile, configIn, 0755)
+			out.MaybeDie(err, "error writing backup config %q: %v", backupFile, err)
+
+			err = afero.WriteFile(fs, configFile, configOut, 0755)
+			out.MaybeDie(err, "error writing config: %v", err)
+
+			log.Infof("Rewrote configuration file %q", configFile)
```

Comment:
The `log` lines in this command are only printed with `-v`, vs. `fmt` which will print by default. I think we'd rather have these printed by default?

Reply:
They're coming out by default when testing here (no `-v`).  I think the logger is configured to print >=Info level messages by default?

Reply:
In this case the `log` library has a bunch of hidden magic formatting in their global logger that we have to setup in `main`, and I can't remember if we actually do that -- vs fmt being wysiwyg